### PR TITLE
Depend on package LibTorchCAPI_jll

### PR DESCRIPTION
**Pull request by KDr2**
The C library has been decoupled to LibTorchCAPI_jll:
- https://github.com/JuliaPackaging/Yggdrasil/pull/3172
- https://github.com/JuliaBinaryWrappers/LibTorchCAPI_jll.jl

### CODE REVIEW

**Comment by KDr2** [comment]
https://github.com/JuliaRegistries/General/tree/master/L/LibTorchCAPI_jll

Actually, it is available now, and all the test cases pass on my local machine, maybe in github actions, it uses a cached Registry? Let's wait a day or so.

File: deps/build.jl [hunk 1]
```diff
@@ -1,78 +0,0 @@
-const PROJECT_DIR = (@__DIR__) |> dirname
```
Comment (yebai) [inline deps/build.jl:1]: Maybe keep this build script as a backup for failing to download binaries from `JuliaBinaryWrappers/LibTorchCAPI_jll.jl`? It's also useful for macOS at the moment

**Review by yebai — CHANGES_REQUESTED** [review]
thanks, @KDr2 - looks good overall, I left one minor comment below.

File: deps/build.jl (hunk 1 above)
Comment (KDr2) [reply deps/build.jl:1]: Update, it can be built on macOS now.

**Pull request merged**